whelk-core/src/main/groovy/whelk/JsonLd.groovy

```diff
@@ -1209,7 +1209,9 @@ class JsonLd {
                 obj = fullObj ? [:] + fullObj : null
             }
             if (obj) {
-                return embed(oId, obj, idMap, new HashSet<String>(embedChain))
+                return depth < 2
+                        ? embed(oId, obj, idMap, new HashSet<String>(embedChain), depth + 1)
```

Comment:
This counts all depths equally, right? It doesn't take e.g. `:QualifiedRole`- or `:StructuredValue`-based entities into account, nor `:category :integral` on the properties? I suspect `2` may be too low in general, since we do have chains like `relatedTo.instanceOf.hasTitle.mainTitle` (and various `instanceOf`-indirected important parts (`subject` et al))... :thinking: